### ISSUE TITLE
CLI tests fail on Windows

### ISSUE THREAD

**Issue by rolando** [LuminosoInsight/python-ftfy#74]
This was found while building conda packages on Windows (py35, py36). Full log here: https://ci.appveyor.com/project/conda-forge/ftfy-feedstock/build/1.0.4/job/d6yr2io539ouweqf

```
======================================================================
ERROR: test_cli.test_stdin
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\bld\ftfy_1494727628610\_t_env\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\bld\ftfy_1494727628610\test_tmp\tests\test_cli.py", line 53, in test_stdin
    output = get_command_output(['ftfy'], stdin=infile)
  File "C:\bld\ftfy_1494727628610\test_tmp\tests\test_cli.py", line 23, in get_command_output
    return subprocess.check_output(args, stdin=stdin, stderr=subprocess.STDOUT, timeout=5).decode('utf-8')
  File "C:\bld\ftfy_1494727628610\_t_env\lib\subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "C:\bld\ftfy_1494727628610\_t_env\lib\subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['ftfy']' returned non-zero exit status 1.
======================================================================
```

Linux/OSX tests passes OK.

**Comment by rspeer** [comment]
Okay. I don't really have a Windows dev machine to test on, so I have to guess a bit at what's going on.

The fact that it got to `test_stdin` means that the other CLI tests passed, so it's likely that the problem relates particularly to what `sys.stdin` is doing on Windows. But I don't know what the problem is.

http://stackoverflow.com/questions/2850893/reading-binary-data-from-stdin/38939320#38939320 tells me that `sys.stdin.buffer` *should* work on Py3 on Windows. Is this not the case?

Would you be able to tell me what actually happens if you run `ftfy` with no arguments on Windows, and give it some text on standard in?

**Comment by rspeer** [comment]
Oh, now I see from the appveyor link that the other CLI tests are also failing.

Is it unreasonable to assume during testing that `ftfy` is installed in the current environment, and is therefore on the path?

**Comment by rolando** [comment]
During the `conda` build, first the package is build, then installed in a test environment and finally tests are executed.

I got this when running `ftfy` directly:
```
> ftfy tests/halibote.txt 
ftfy error:
Unfortunately, this output stream does not support Unicode.
You might be trying to output to the Windows Command Prompt (cmd.exe), which
does not fully support Unicode for historical reasons. In general, we recommend
finding a way to run Python without using cmd.exe.
You can work around this problem by using the '-o filename' option in ftfy to
output to a file instead.
```

**Comment by rolando** [comment]
I was able to fix the issue by using the `PYTHONIOENCODING` envvar.

```
(C:\bld\ftfy_1494873422494\_t_env) (root) C:\bld\ftfy_1494873422494\test_tmp>SET "PYTHONIOENCODING=utf8"   & ftfy tests/halibote.txt 
【更新】《哈利波特》石堧卜才新婚娶初戀今痠逝
```

**Comment by rolando** [comment]
The environment variable fixes the command output but tests still fail due to line separator `\r\n`:
```
======================================================================
FAIL: test_cli.test_stdin
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\bld\ftfy_1494876198838\_t_env\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\bld\ftfy_1494876198838\test_tmp\tests\test_cli.py", line 54, in test_stdin
    eq_(output, CORRECT_OUTPUT)
AssertionError: '【更新】《哈利波特》石堧卜才新婚娶初戀今痠逝\r\n' != '【更新】《哈利波特》石堧卜才新婚娶初戀今痠逝\n'
----------------------------------------------------------------------
```

Full log https://ci.appveyor.com/project/rolando/ftfy-feedstock/build/job/fvoxb48ayd1t8rjf

**Comment by rolando** [comment]
Fixed by #76
